src/main/java/ro/msg/learning/shop/controller/LoginController.java

```diff
@@ -0,0 +1,40 @@
+package ro.msg.learning.shop.controller;
+
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@Controller
+public class LoginController {
```

Comment:
To be honest, I am not sure why you need this controller. Spring Security has built-in routes for login and logout (for Form based authentication).